src/controllers/legacycontrollermapping.h

```diff
@@ -1,14 +1,26 @@
 #pragma once
 
+#include <util/assert.h>
+
 #include <QDebug>
 #include <QDir>
+#include <QDomElement>
 #include <QHash>
 #include <QList>
 #include <QSharedPointer>
 #include <QString>
+#include <QtGlobal>
 #include <memory>
+#include <vector>
 
+#include "controllers/legacycontrollersettings.h"
+#include "controllers/legacycontrollersettingslayout.h"
 #include "defs_urls.h"
+#include "preferences/usersettings.h"
+
+#define CONTROLLER_SETTINGS_PREFERENCE_GROUP_KEY "[ControllerSettings_%1_%2]"
```

Comment:
This can be also moved to the anonymous space and become a const QString initialized by a QStringLiteral()